### PR TITLE
Finished configuring devise. Able to sign_up sigin and sign out

### PR DESCRIPTION
**Pull request by sharadalt**

### CODE REVIEW

**Comment by eliotsykes** [comment]
Hello this is a test comment

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    <%= f.label :name %>
+    <%= f.text_field :name, class: 'form-control' %>
+  </div>
+
+  <div class="form-group">
+    <%= f.label :email %>
+    <%= f.email_field :email, class: 'form-control' %>
+  </div>
+
+  <div class="form-group">
```
Comment (eliotsykes) [inline app/views/devise/registrations/new.html.erb]: Another test comment

File: app/assets/stylesheets/welcome.scss [hunk 2]
```diff
@@ -1,3 +1,39 @@
...
+  margin-right: 600px ;
+}
+
+h2 {
+  color:green;
+}
+
+h3 {
+  color:green;
+}
```
Comment (eliotsykes) [inline app/assets/stylesheets/welcome.scss:39]: As the style for h2 and h3 are identical, lines 33 to 39 can be reduced to:

``` css
h2, h3 {
  color: green;
}
```

File: app/assets/stylesheets/welcome.scss [hunk 3]
```diff
@@ -1,3 +1,39 @@
...
+  width: 20%;
+}
+
+.posts {
+  border: 3px solid green ;
+  margin-right: 250px;
+  margin-left: auto;
+  padding: 5px 10px 10px 5px;
+  width: 20%;
+}
```
Comment (eliotsykes) [inline app/assets/stylesheets/welcome.scss:26]: @sharadalt 
For all HTML elements that you'd like to have the green border, consider writing a dedicated style class that can be reused. For example, here is `border-medium` css class that can be reused on any HTML element:

``` css
.border-medium {
  border: 3px solid green;
}
```

Then to apply `border-medium` in HTML:

``` html
<section class="border-medium">....</section>
<div class="boxed border-medium">...</div>
<p class="posts border-medium">...</p>
```

The above elements will all have the 3px green border.

In addition, this means the `border: 3px solid green;` line can be removed from the CSS at lines 5, 12, and 21 above in `welcome.scss`:

``` diff
.boxed {
-  border: 3px solid green;
...
.section {
-  border: 3px solid green;
...
.posts {
-  border: 3px solid green;
```

As always please let me know if you've got questions.

Comment (eliotsykes) [inline app/assets/stylesheets/welcome.scss:26]: @sharadalt PS. I recommend viewing **all** feedback on GitHub directly - the emails aren't very good and don't show the lines I'm referring to. https://github.com/sharadalt/blocipedea/pull/1/files

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -1,15 +1,27 @@
...
   <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
   <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
   <%= csrf_meta_tags %>
 </head>
 <body>
+  <div class="container">
+    <ul class="nav nav-tabs">
+      <div class="pull-right user-info">
+        <br>
+        <P ALIGN="center">
```
Comment (eliotsykes) [inline app/views/layouts/application.html.erb:14]: Try to use lowercase HTML tags and avoid using inline HTML styles like `ALIGN=...`. Try to keep all styles in the stylesheet files (i.e. `.scss` files).

``` diff
-<P ALIGN="center">
+<p>
```

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -1,15 +1,27 @@
...
+            <% if user_signed_in? %>
+              <%= current_user.name %> <%= link_to "Sign Out", destroy_user_session_path, method: :delete %>
+            <% else %>
+              <%= link_to "Sign In", new_user_session_path %>
+            <% end %>
+        </P>
```
Comment (eliotsykes) [inline app/views/layouts/application.html.erb:20]: Also lowercase closing HTML tags for consistency

``` diff
-</P>
+</p>
```

File: app/views/welcome/index.html.erb [hunk 6]
```diff
@@ -1,11 +1,30 @@
+<br>
+
+
 <% content_for :top_content do %>
+  <CENTER>
```
Comment (eliotsykes) [inline app/views/welcome/index.html.erb:5]: Try not to use `<center>` tags as these are a styling concern and a deprecated technique for centering.

To align text centrally in a heading, use the stylesheets:

``` css
h2 {
  text-align: center;
}
```

File: app/views/welcome/index.html.erb [hunk 7]
```diff
@@ -1,11 +1,30 @@
...
-      <p>
-        <%= link_to 'Register', new_user_registration_path, class: 'btn btn-primary btn-lg' %>
-      </p>
+      <div class="boxed">
+        <h3>Create and Share Markdown Wikis</h3>
+          <h3>
+            <%= link_to 'Sign Up', new_user_registration_path, class: 'btn btn-primary btn-lg' %>
+          </h3>
+        </div>
+          <CENTER>
```
Comment (eliotsykes) [inline app/views/welcome/index.html.erb:16]: Try to do without the `<CENTER>` tag, favor using stylesheets for styling and layout concerns.

File: config/initializers/devise.rb [hunk 8]
```diff
@@ -6,13 +6,13 @@
   # confirmation, reset password and unlock tokens in the database.
   # Devise will use the `secret_key_base` on Rails 4+ applications as its `secret_key`
   # by default. You can change it below and use your own secret key.
-  config.secret_key = '6fac83e00a97146fc3f098cbaffdd60758a0912e4dbb3d726e29cf0fc72689a594e9a102005ba0ee647093b24ec425a23f1703aa42ce2dfc15e8d8de4b459fc5'
-
+  config.secret_key = '6fac83e00a97146fc3f098cbaffdd60758#a0912e4dbb3d726e29cf0fc72689a594e9a102005ba0ee647093b24ec425a23f1703aa42ce2dfc15e8d8de4b459fc5'
```
Comment (eliotsykes) [inline config/initializers/devise.rb:9]: Comment out `config.secret_key`. Devise will use the Rails secret key already defined in `config/secrets.yml`.

File: config/initializers/devise.rb [hunk 9]
```diff
@@ -215,7 +215,7 @@
 
   # Configure the default scope given to Warden. By default it's the first
   # devise role declared in your routes (usually :user).
-  # config.default_scope = :user
+  config.default_scope = :user
```
Comment (eliotsykes) [inline config/initializers/devise.rb:218]: This can be commented out.

File: config/initializers/devise.rb [hunk 10]
```diff
@@ -230,7 +230,7 @@
   # should add them to the navigational formats lists.
   #
   # The "*/*" below is required to match Internet Explorer requests.
-  # config.navigational_formats = ['*/*', :html]
+  config.navigational_formats = ['*/*', :html]
```
Comment (eliotsykes) [inline config/initializers/devise.rb:233]: These can be commented out.

**Comment by eliotsykes** [comment]
Overall looking good for your first time using Devise, well done :+1:

File: config/initializers/devise.rb (hunk 8 above)
Comment (sharadalt) [inline config/initializers/devise.rb:9]: I commented it

File: app/views/welcome/index.html.erb (hunk 7 above)
Comment (sharadalt) [inline app/views/welcome/index.html.erb:16]: Taken care.

File: app/assets/stylesheets/welcome.scss (hunk 3 above)
Comment (sharadalt) [inline app/assets/stylesheets/welcome.scss:26]: The above comment taken care.

File: app/assets/stylesheets/welcome.scss (hunk 2 above)
Comment (sharadalt) [inline app/assets/stylesheets/welcome.scss:39]: The above taken care.

File: app/views/layouts/application.html.erb (hunk 4 above)
Comment (sharadalt) [inline app/views/layouts/application.html.erb:14]: Taken care.

File: app/views/layouts/application.html.erb (hunk 5 above)
Comment (sharadalt) [inline app/views/layouts/application.html.erb:20]: Taken care.

File: app/views/welcome/index.html.erb (hunk 6 above)
Comment (sharadalt) [inline app/views/welcome/index.html.erb:5]: Taken care,

File: config/initializers/devise.rb (hunk 9 above)
Comment (sharadalt) [inline config/initializers/devise.rb:218]: line 218 commented out.

File: config/initializers/devise.rb (hunk 10 above)
Comment (sharadalt) [inline config/initializers/devise.rb:233]: Lines 233 and 236 commented out.

**Pull request merged**